lib/PrintAsClang/_SwiftCxxInteroperability.h

```diff
@@ -279,6 +279,88 @@ class Error {
   void * _Nonnull opaqueValue = nullptr;
 };
 
+/// The Expected class has either an error or an value.
+template<class T>
+class Expected {
+public:
+
+  /// Default
+  constexpr Expected() { has_val = false; }
+  constexpr Expected(const swift::Error& error_val) : err(error_val) { has_val = false; }
+  constexpr Expected(T buf) {
+    new (&buffer) T(buf);
+    has_val = true;
+  }
+
+  /// Copy
+  constexpr Expected(Expected const& other) noexcept {
+    if (other.has_value())
+      new (&buffer) T(other.value());
+    else
+      new (&err) Error(other.error());
+
+    has_val = other.has_value();
+  }
+  /// Move
+  // FIXME: Implement move semantics when move Swift values is possible
+  constexpr Expected(Expected&&) noexcept { abort(); }
+
+  ~Expected() noexcept { }
+
+  /// assignment
+  constexpr auto operator=(Expected&& other) noexcept { auto temp = buffer; buffer = other.buffer; other.buffer = temp; }
+  constexpr auto operator=(Expected&) noexcept = delete;
+
+  /// For accessing T's members
+  // precondition: has_value() == true
+  constexpr T const *_Nonnull operator->() const noexcept { return reinterpret_cast<const T *>(buffer); }
```

Comment:
Can we check `has_value` explicitly like in `value` instead of specifying the precondition in the comments?

Reply:
Yes!

Reply:
if it doesn't has a value can we abort?

Reply:
yes.